src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Generators/CSharpIVsSingleFileGeneratorFactoryAggregator.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Packaging;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.ProjectSystem.VS.Utilities;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Generators
+{
+    [Export(ExportContractNames.VsTypes.ProjectNodeComExtension)]
+    [AppliesTo(ProjectCapability.CSharp)]
+    [ComServiceIid(typeof(IVsSingleFileGeneratorFactory))]
+    class CSharpIVsSingleFileGeneratorFactoryAggregator : IVsSingleFileGeneratorFactoryAggregator
+    {
+        [ImportingConstructor]
+        public CSharpIVsSingleFileGeneratorFactoryAggregator([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
```

Comment:
Instead of C# and VB components, can we please drive this off a project property in the design-time targets? We drive differentiation between languages from that instead of code wherever possible.